src/fflogs.ts

```diff
@@ -0,0 +1,228 @@
+import {AxiosRequestConfig} from 'axios'
+
+// -----
+// Fight
+// -----
+
+export interface Fight {
+	id: number
+
+	boss: number
+	name: string
+	zoneID: number
+	zoneName: string
+	lastPhaseForPercentageDisplay?: number
+
+	difficulty?: number
+	bossPercentage?: number
+	fightPercentage?: number
+	kill?: boolean
+	partial?: number
+	size?: number
+	standardComposition?: boolean
+
+	start_time: number
+	end_time: number
+}
+
+export interface Phase {
+	boss: Fight['boss']
+	phases: string[]
+}
+
+// -----
+// Actors
+// -----
+
+export enum ActorType {
+	// Enemy
+	BOSS = 'Boss',
+	NPC = 'NPC',
+
+	// Friendly
+	PALADIN = 'Paladin',
+	WARRIOR = 'Warrior',
+	DARK_KNIGHT = 'DarkKnight',
+	WHITE_MAGE = 'WhiteMage',
+	SCHOLAR = 'Scholar',
+	ASTROLOGIAN = 'Astrologian',
+	MONK = 'Monk',
+	DRAGOON = 'Dragoon',
+	NINJA = 'Ninja',
+	SAMURAI = 'Samurai',
+	BARD = 'Bard',
+	MACHINIST = 'Machinist',
+	BLACK_MAGE = 'BlackMage',
+	SUMMONER = 'Summoner',
+	RED_MAGE = 'RedMage',
+	LIMIT_BREAK = 'LimitBreak',
+
+	// Pet
+	PET = 'Pet'
+}
+
+export interface ActorFightInstance {
+	id: Fight['id'],
+	groups?: number,
+	instances?: number,
+}
+
+export interface Actor {
+	guid: number
+	id: number
+	name: string
+	type: ActorType
+	fights: ActorFightInstance[]
+}
+
+export interface Pet extends Actor {
+	petOwner: Actor['id']
+}
+
+export interface ActorResources {
+	hitPoints: number
+	maxHitPoints: number
+	mp: number
+	maxMP: number
+	tp: number
+	maxTp: number
+	x: number
+	y: number
+}
+
+// -----
+// Events
+// -----
+
+export enum AbilityType {
+	PHYSICAL_DOT = 1,
+	HEAL = 8,
+	SPECIAL = 32,
+	MAGICAL_DOT = 64,
+	PHYSICAL_DIRECT = 128,
+	LIMIT_BREAK = 256,
+	MAGICAL_DIRECT = 1024,
+}
+
+export interface Ability {
+	abilityIcon: string
+	guid: number
+	name: string
+	type: AbilityType
+}
+
+export interface Event {
+	timestamp: number
+	type: string
+
+	sourceID: number
+	sourceIsFriendly: boolean
+	targetID: number
+	targetInstance: number
+	targetIsFriendly: boolean
+
+	// Hell if I know _every_ event key - cop out for anything not explicitly set
+	[key: string]: any
```

Comment:
If I get motivated enough I'll just make a PR later to fix this... 😖 

FWIW, in my other codebases no `any` are allowed at all unless it really is an incomplete definition, which are normally marked by a `type TODO = any` instead

Reply:
Despite the comment, the event types are actually near-complete. Just needs some tweaks to filter handling before I can remove the index type.